src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.references;
+
+import net.mcreator.blockly.data.BlocklyXML;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.Sound;
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.generator.GeneratorWrapper;
+import net.mcreator.generator.mapping.MappableElement;
+import net.mcreator.minecraft.DataListEntry;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.workspace.IWorkspaceProvider;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.SoundElement;
+import net.mcreator.workspace.resources.Model;
+import net.mcreator.workspace.resources.TexturedModel;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.lang.reflect.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+public class ReferencesFinder {
+
+	public static List<ModElement> searchModElementUsages(Workspace workspace, ModElement element) {

```

Comment:
I think all those methods should return Set<ModElement> and add things to HashSet as getting the same reference multiple times does no good and in example of ProcedureGUI will run regenerate multiple times if same element requires one procedure multiple times, for example.